entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Complex Query Operators - EF Core
+author: smitpatel
+ms.date: 10/03/2019
+ms.assetid: 2e187a2a-4072-4198-9040-aaad68e424fd
+uid: core/querying/complex-query-operators
+---
+# Complex Query Operators
+
+Language Integrated Query (LINQ) contains many complex operators, which combine multiple data sources or does complex processing. Not all LINQ operators have suitable translation in the server side. Sometimes, query in one form translates to server but if written in different form doesn't translate even if the result is same. This page describes some of the complex operators and their supported variations. In future, we may recognize more pattern and add their translations. It's also important to keep in mind that translation support varies between providers. A particular query, which is translated in SqlServer, mayn't work for SQLite database.
+
+> [!TIP]
+> You can view this article's [sample](https://github.com/aspnet/EntityFramework.Docs/tree/master/samples/core/Querying) on GitHub.
+
+## Join
+
+LINQ Join operator allows you to connect two data sources based on the key selector on each source, generating a tuple of values when key matches. It naturally translates to `INNER JOIN` on relational databases. While LINQ Join has outer and inner key selectors, database requires a single join condition. So EF Core will generate join condition by comparing outer key selector to inner key selector for equality. Further, if key selectors are anonymous types, EF Core generates a join condition to do equality component wise.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#Join)]
+
+```SQL
+SELECT [p].[PersonId], [p].[Name], [p].[PhotoId], [p0].[PersonPhotoId], [p0].[Caption], [p0].[Photo]
+FROM [PersonPhoto] AS [p0]
+INNER JOIN [Person] AS [p] ON [p0].[PersonPhotoId] = [p].[PhotoId]
+```
+
+## GroupJoin
+
+LINQ GroupJoin operator allows you to connect two data sources in similar to Join but it creates a group of inner values for matching outer elements. Executing query like following generates result of `Blog` & `IEnumerable<Post>`. Since databases (especially relational databases) don't support representing a collection of client-side objects, GroupJoin can't be translated to server in many cases. It requires to get all the data from server to do GroupJoin without special selector (first query below). But if the selector is limiting data being selected then fetching all data from server may cause performance issues (second query below). That's why EF Core doesn't translate GroupJoin.
```

Comment:
```suggestion
LINQ's GroupJoin operator allows you to connect two data sources similar to Join, but it creates a group of inner values for matching outer elements. Executing a query like the following generates a result of `Blog` & `IEnumerable<Post>`. Since databases--especially relational databases--don't have a way to represent collections of client-side objects, GroupJoin can't be translated to the server in many cases. It requires you to get all of the data from the server to do GroupJoin without a special selector (first query below). But if the selector is limiting data being selected then fetching all of the data from the server may cause performance issues (second query below). That's why EF Core doesn't translate GroupJoin.
```